File: cpp/gabaritos/gabarito28.cpp
#include <iostream>
#include <string>
#include <regex>

bool validar_email(const std::string& email) {
    std::regex pattern(R"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)");  
    return std::regex_match(email, pattern);
}

int main() {
    std::string email;
    std::cout << "Digite um email: ";
    std::getline(std::cin, email);
    if (validar_email(email)) {
        std::cout << "Email válido" << std::endl;
    } else {
        std::cout << "Email inválido" << std::endl;
    }
    return 0;
}

File: csharp/gabaritos/gabarito28.cs
using System;
using System.Text.RegularExpressions;

class Program
{
    static bool ValidarEmail(string email)
    {
        string pattern = @"^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$";
        return Regex.IsMatch(email, pattern);
    }

    static void Main()
    {
        Console.Write("Digite um email: ");
        string email = Console.ReadLine();
        if (ValidarEmail(email))
        {
            Console.WriteLine("Email válido");
        }
        else
        {
            Console.WriteLine("Email inválido");
        }
    }
}